core/src/main/scala/kafka/admin/ReassignPartitionsCommand.scala

```diff
@@ -314,31 +319,51 @@ class ReassignPartitionsCommand(zkUtils: ZkUtils, partitions: Map[TopicAndPartit
     }
   }
 
-  def addThrottledReplicaList(): Unit = {
-    //apply the throttle to all move destinations and all move sources
-    val existing = zkUtils.getReplicaAssignmentForTopics(partitions.map(_._1.topic).toSeq)
-    val moves = replicaMoves(existing, proposed = partitions)
-    for (topic <- partitions.keySet.map(tp => tp.topic).toSeq.distinct)
-      AdminUtils.changeTopicConfig(zkUtils, topic, new Properties {
-        put(LogConfig.ThrottledReplicasListProp, moves.get(topic).get )
-      })
-    println(s"Throttles were added to the following replicas: $moves")
+  def assignThrottledReplicas(): Unit = {
+    val topicsForMove = partitions.map { case (tp, _) => tp.topic }.toSeq
+    val current = zkUtils.getReplicaAssignmentForTopics(topicsForMove)
+    assignThrottledReplicas(current, partitions)
+  }
+
+  private[admin] def assignThrottledReplicas(allExisting: Map[TopicAndPartition, Seq[Int]], allProposed: Map[TopicAndPartition, Seq[Int]], admin: AdminUtilities = AdminUtils): Unit = {
+    for (topic <- allProposed.keySet.map(tp => tp.topic).toSeq.distinct) {
+      val (existing, proposed) = filterBy(topic, allExisting, allProposed)
```

Comment:
The input can cover a range of topics, as we can be rebalancing many topics at once, and the logic is a bit simpler if we do it a topic at a time. 

To put it another way, if we don't filter here the test shouldFindMovingReplicasMultipleTopics would fail.

Let me know if I'm missing your point.